### ISSUE TITLE
Production build number

### ISSUE THREAD

**Issue by matthewp** [stealjs/steal#1035]
@justinbmeyer has expressed interest in Steal providing a build number of some sort that can be used by libraries. This issue is to discuss this issue and how it can be implemented.

One constraint is that this feature shouldn't block browser caching, so appending a number into the bundles won't work.

**Comment by justinbmeyer** [comment]
@matthewp Can you remind me why that wouldn't work again?

**Comment by matthewp** [comment]
Note that I'm talking about the idea to put a build number into the generated `steal.production.js`. This wouldn't work because it would cause steal.production.js to not come from the cache.

**Comment by pYr0x** [comment]
what is about this idea:
- untouch the `steal.prduction.js`
- on build create a `bundles.json` file that contain all the bundles that properly will write into the bundled files like 
```
/*[system-bundles-config]*/
System.bundles = {"bundles/access-index":["triwala-vk@1.0.0#util"]};
```
put this file on the dist folder right next to the `steal.production.js`
- on production env lets do this script tag
`<script src="dist/steal.production.js" main="my-app/main" bundles="bundles.json"></script>`

steal will load this bundle file first. then load the `my-app/main` file.

the idea of production-number is to put a hash on each js file

**Comment by justinbmeyer** [comment]
What if the build number was written into the main?  Yeah, you'd have to load the main even if there weren't any other changes in it, but I don't think that matters practically so much.

**Comment by matthewp** [comment]
The main bundle is usually the largest bundle, it's the one that you want cached the most.

@justinbmeyer is there a bug in can-connect that describes the need for this type of functionality? I'd like to understand the use-case better.

**Comment by justinbmeyer** [comment]
@matthewp I guess my point would be that on a new release, you've almost certainly changed the main bundle.  That having it change with the build number would overlap so often that the times it didn't might be ok forcing everyone to get a new version.

**Comment by matthewp** [comment]
It's definitely possible to change only 1 page of your application which would then result in only that bundle changing. I don't know what percentage of the time that is, but I wouldn't want my main bundle to skip the cache in that case.

**Comment by pYr0x** [comment]
@matthewp i want to work on this feature like i described in my comment. can you confirm that this is a way it could / should work?

**Comment by matthewp** [comment]
No, I don't think so because it would mean you'd have to wait for this bundle config to load before loading the main bundle. Which would cause the app to load slower.

I'd really like to learn more about the use-case before we proceed with a sub-optimal solution. I don't want to do anything that causes slower production loads unless there is a very strong reason for it.
